lib/pbench/cli/agent/commands/results/move.py

```diff
@@ -0,0 +1,196 @@
+
+
+class MoveResults(BaseCommand):
+    """
+        TODO:  This is latent code -- it is currently unused and largely
+            untested, intended to implement a future tool to replace
+            pbench-make-result-tb and/or pbench-move-results.
+    """
+
+    def __init__(self, context: click.Context):
+        super().__init__(context)
+
+    def execute(self, copy_only=False) -> int:
+        logger = get_pbench_logger("pbench-agent", self.config)
+
+        temp_dir = tempfile.mkdtemp(
+            dir=self.config.pbench_tmp, prefix="pbench-move-results."
+        )
+
+        runs_copied = 0
+        failures = 0
+        no_of_tb = 0
+
+        for dirent in self.config.pbench_run.iterdir():
+            if not dirent.is_dir():
+                continue
+            if dirent.name.startswith("tools-") or dirent.name == "tmp":
+                continue
+
+            no_of_tb += 1
+            result_dir = dirent
+
+            try:
+                mrt = MakeResultTb(result_dir, temp_dir, self.config, logger)
+            except FileNotFoundError as e:
+                logger.error("File Not Found Error, {}", e)
+                continue
+            except NotADirectoryError as e:
+                logger.error("Bad Directory, {}", e)
+                continue
+
+            try:
+                result_tb_name = mrt.make_result_tb()
+            except BadMDLogFormat as e:
+                logger.warning("Bad Metadata.log file encountered, {}", e)
+                failures += 1
+                continue
+            except FileNotFoundError as e:
+                logger.debug("File Not Found error, {}", e)
+                failures += 1
+                continue
+            except RuntimeError as e:
+                logger.warning("Unexpected Error encountered, {}", e)
+                failures += 1
+                continue
+            except Exception as e:
+                logger.debug("Unexpected Error occurred, {}", e)
+                failures += 1
+                continue
+
+            try:
+                crt = CopyResultTb(
+                    self.context.controller, result_tb_name, self.config, logger
+                )
+            except FileNotFoundError as e:
+                logger.error("File Not Found error, {}", e)
+                failures += 1
+                continue
+
+            try:
+                crt.copy_result_tb(self.context.token)
+            except (FileUploadError, RuntimeError) as e:
+                logger.error("Error uploading a file, {}", e)
+                failures += 1
+                continue
+
+            try:
+                # We always remove the constructed tar ball, regardless of success
+                # or failure, since we keep the result directory below on failure.
+                os.remove(result_tb_name)
+            except OSError:
+                logger.error("Failed to remove {}", result_tb_name)
+                failures += 1
+                continue
```

Comment:
I suspect that we should not count an exception at line 97 as a "failure" at line 100 -- after all, the result _did_ get copied....

Reply:
Ditto lines 109 and 118.

Reply:
Existing code ...

Reply:
[Not really](https://github.com/distributed-system-analysis/pbench/pull/2325#discussion_r636139655) ...

At best, it's a _port_ of existing code.  The question remains, do we, going forward, want it to do what it is currently structured to do?